File: 2405. Optimal Partition of String/set_1.py
class Solution:
    def partitionString(self, s: str) -> int:
        answer = 1 # start with 1 as all characters might be unique
        character_set = set()
        for index in range(len(s)):
            if s[index] in character_set:
                answer += 1
                character_set = set()
            character_set.add(s[index])
        return answer
    
# time O(n)
# space O(26)

File: 2405. Optimal Partition of String/set_1.cpp
# include <string>

class Solution {
public:
    int partitionString(string s) {
        int partitions = 1;
        unordered_set<char> character_set;
        for (int index = 0; index < s.length(); index++){
            if (character_set.find(s[index]) != character_set.end()){
                partitions += 1;
                character_set.clear();               
            }
            character_set.insert(s[index]);
        }
        return partitions;
    }
};